src/DataProtection/DataProtection/src/TypeForwardingActivator.cs

```diff
@@ -64,6 +62,26 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
         }
 
         protected string RemoveVersionFromAssemblyName(string forwardedTypeName)
-            => _versionPattern.Replace(forwardedTypeName, "");
+        {
+            // Type, Assembly, Version={Version}, Culture={Culture}, PublicKeyToken={Token}
```

Comment:
Could we use `AssemblyName` to do this?


```suggestion
            try
            {
               var parsedName = new AssemblyName(typeof(Program).Assembly.FullName) { Version = null };
               return parsedName.ToString();
            }
            catch
            {
               return forwardedTypeName;
            }
```

Reply:
No, you're need to slice off the type name first. I'm considering another approach where I store a list of provider types by type name and assembly as well